doc/diagnostics/VMState.md

```diff
@@ -0,0 +1,88 @@
+<!--
+Copyright (c) 2018, 2018 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+**VM state**
+
+The VM state is a numeric field stored in J9VMThread->omrVMThread->vmState that uses 32-bits to encode the current state of the thread. See the `J9VMSTATE_` defined constants in the code. For example, if the thread is currently JIT compiling a method, the VM state indicates which compilation phase the compiler is in. If the VM crashes for any reason, the VM state is reported via standard error and in the javacore dump (in which it is referred to as "VM flags"), and serves as a first-order indicator of the failing component during problem determination. 
+
+**Decoding**
+
+The high order half of the 32-bit VM state indicates the component.
+
+| Value  | Component |
+| -----  | :-------: |
+| 0x0000 | none |
+| 0x0001 | interpreter |
+| 0x0002 | garbage collector |
+| 0x0003 | stack growth |
+| 0x0004 | JNI code |
+| 0x0005 | JIT compilation |
+| 0x0006 | bytecode verifier |
+| 0x0007 | runtime verifier |
+| 0x0008 | shared class cache |
+| 0x0011 | sniff-n-whack stack validator |
+
+The VM includes a `-Xjit:vmstate=` command line option to decode a VM state, see the following example. Note the "0x" and the three leading zeros are required.
```

Comment:
It's been sort of an unwritten guideline with the documentation in the `compiler` folder that all lines only be ~70 characters long, since in markdown, putting a single new line doesn't change the way it is rendered (eg see https://raw.githubusercontent.com/eclipse/openj9/master/doc/compiler/MemoryManager.md vs https://github.com/eclipse/openj9/blob/master/doc/compiler/MemoryManager.md). Perhaps this guideline should be made consistent through OpenJ9?

Reply:
I've wrapped lines at ~70 columns where feasible.

Reply:
In addition to readability, another reason we wrapped lines this way was to make reviews such as this one easier to do.  A reviewer can attach their comment closer to the context in which it applies, which is harder to do on a line that wraps across multiple lines.